scrapers/scraper.py

```diff
@@ -0,0 +1,267 @@
+        stats = json.loads(requests.get("https://apertium.projectjj.com/stats-service/apertium-{}/Lexc".format(language)).text)["stats"]
+        for statistic in stats:
+            if statistic["stat_kind"] == "Stems":
+                stems = statistic["value"]
+                break
+        for topic in package["topics"]:
+            if topic[len("apertium-"):] in langLocations:
+                location = topic[len("apertium-"):]
+                break
+
+        subprocess.call(['git', 'clone', 'https://github.com/apertium/apertium-{}'.format(language)], stderr=subprocess.DEVNULL)
+        shutil.copyfile("./.mailmap", "./apertium-{}/.mailmap".format(language))
+        lines = subprocess.check_output(['git', 'log', '--format=%aN', '--follow', 'apertium-{0}.{0}.lexc'.format(language)], cwd='./apertium-{}/'.format(language)).decode('utf-8')
+
+        if updatemailmap:
+            emails = subprocess.check_output(['git', 'log', '--format=<%aE>', '--follow', 'apertium-{0}.{0}.lexc'.format(language)], cwd='./apertium-{}/'.format(language)).decode('utf-8').split("\n")
+            mailmap = open("./.mailmap", "r", encoding="utf-8").read()
+            for email in emails:
+                if not email in mailmap:
+                    print(email)
+
+        shutil.rmtree("./apertium-{}/".format(language), onerror=del_rw)
+
+        authors = lines.split("\n")
+        authors.pop() #last line is always empty
+        contributors = []
+        authorCount = Counter(authors)
+        for contributor, count in authorCount.items():
+            contributors.append({"user": contributor, "value": count})
+
+        wikiData = requests.get(wikiURL).text
+        rows = etree.fromstring(wikiData, parser=etree.XMLParser(encoding="utf-8"))[1][0][0][0][3][3][8]
+        for row in rows[2:]:
+            if row[0][0][0].text[len("apertium-"):] == language:
+                state = row[6].text.strip()
+                break
+            state = "unknown"
+
+        data.append({
+            "lang": language,
+            "state": state,
+            "stems": stems,
+            "location": "apertium-{} ({})".format(language, location),
+            "contributors": contributors
+        })
+
+    return data
+
+
+def pairData(packages):
+    """Returns the locations and stems of all specified pairs"""
+    data = []
+    for package in packages:
+        if not re.match(r"apertium-\w+-\w+", package["name"]):
+            continue
+
+        pairName = package["name"].replace("apertium-", "")
+        pairSet = set(pairName.split("-"))
+        if not pairSet <= set(languages):
+            continue
+        for topic in package["topics"]:
+            if topic[len("apertium-"):] in pairLocations:
+                location = topic[len("apertium-"):]
+                break
+        stems = json.loads(requests.get("https://apertium.projectjj.com/stats-service/apertium-{}/bidix".format(pairName)).text)["stats"][0]["value"]
```

Comment:
Why bother using stats-service if it won't give you the history?

Reply:
Right here, it only needs the current amount of stems, so it's much easier to use the already calculated ones from stats-service

Reply:
Hmm, okay. Are there different modes to get latest vs update?

Reply:
Yes, check my original comment on this PR

Reply:
gotcha